roles/openshift_manageiq/tasks/main.yaml

```diff
@@ -59,6 +59,9 @@
   register: oshawkular_create_cluster_role
   failed_when: "'already exists' not in oshawkular_create_cluster_role.stderr and oshawkular_create_cluster_role.rc != 0"
   changed_when: oshawkular_create_cluster_role.rc == 0
+  # AUDIT:changed_when_note: Checking the return code is insufficient
+  # here. We really need to verify the if the role even exists before
+  # we run this task.
```

Comment:
I suppose that does work.

The audit note isn't so much about checking that the task worked correctly but more about writing the task in a more idempotent way.

The way that task is written now it will **always** run. That's OK because if the role already exists there are no side-effects, **BUT** it's still another task that has to run. We **should** expect to see this task result as `skipped` because nothing has to happen.

In conclusion, @detiber , it's an audit note for future idempotency improvements. The task works fine the way it is but this is not *the ideal way* to do this.